xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -597,7 +598,7 @@ BuildObject(CFileItem&                    item,
 
         /* Get the number of children for this container */
         if (with_count && upnp_server) {
-            if (object->m_ObjectID.StartsWith("virtualpath://")) {
+            if (StringUtils::StartsWithNoCase(DecodeObjectId(object->m_ObjectID.GetChars()), "virtualpath://")) {
```

Comment:
Can we introduce a temporary variable for the decoded objectID? It is difficult to read the way it is now.